builder/builder.go

```diff
@@ -58,81 +58,115 @@ func getAuthData(filename string) (*dockerAuthData, error) {
 }
 
 // Build Docker Image.
-func (imgBuilder ImageBuilder) BuildImage() error {
+func (imgBuilder ImageBuilder) BuildImage(assignmentEnv bool) error {
 
-	buildContext := context.Background()
+	backgroundContext := context.Background()
 	dockerClient, err := client.NewEnvClient()
 	if err != nil {
 		log.Fatalf("error in creating a docker client: %v", err)
+		return err
+	}
+
+	// Create a build context tar for the image.
+	dockerFilepath := imgBuilder.dockerFilename
+
+	dockerBuildContext, err := getDockerBuildContextTar(dockerFilepath)
+	if err != nil {
+		log.Fatalf("error in creating a docker client: %v", err)
+		return err
 	}
 
-	// Create a tar for build context.
-	tarBuffer := new(bytes.Buffer)
-	writer := tar.NewWriter(tarBuffer)
+	tagName := getTagName(imgBuilder, assignmentEnv)
+
+	response, err := dockerClient.ImageBuild(
+		backgroundContext,
+		dockerBuildContext,
+		types.ImageBuildOptions{
+			Dockerfile: dockerFilepath,
+			Tags:       []string{tagName}})
+	if err != nil {
+		log.Fatalf("unable to build docker image: %v", err)
+		return err
+	}
 	defer func() {
-		err = writer.Close()
+		err = response.Body.Close()
 		if err != nil {
 			log.Fatal(err)
 			return
 		}
 	}()
+	_, err = io.Copy(os.Stdout, response.Body)
+	if err != nil {
+		log.Fatalf("unable to read image build response: %v", err)
+		return err
+	}
 
-	dockerFilepath := imgBuilder.dockerFilename
+	return nil
+}
+
+// Get Tag name for building image.
+func getTagName(imgBuilder ImageBuilder, assignmentEnv bool) string {
+	var tagName string
+	if assignmentEnv {
+		tagName = fmt.Sprintf("%s/%s", imgBuilder.authData.Username, imgBuilder.LanguageImageFormat)
+	} else {
+		tagName = fmt.Sprintf("%s/%s:%s", imgBuilder.authData.Username,
+			imgBuilder.authData.Repository, imgBuilder.authData.Version)
+	}
+
+	return tagName
+}
+
+// Get Docker Build Context Tar Reader for building image.
+func getDockerBuildContextTar(dockerFilepath string) (*os.File, error) {
 	dockerFileReader, err := os.Open(dockerFilepath)
 	if err != nil {
 		log.Fatalf(" unable to open dockerfile: %v", err)
 	}
-	readDockerFile, err := ioutil.ReadAll(dockerFileReader)
+	fileInfo, err := os.Stat(dockerFilepath)
 	if err != nil {
-		log.Fatalf("unable to read dockerfile: %v", err)
+		return nil, err
 	}
 
-	tarHeader := &tar.Header{
-		Name: dockerFilepath,
-		Size: int64(len(readDockerFile)),
-	}
-	err = writer.WriteHeader(tarHeader)
+	buildContextTar := new(archivex.TarFile)
+	err = buildContextTar.Create(constants.BuildContextTar)
 	if err != nil {
-		log.Fatalf("unable to write tar header: %v", err)
+		return nil, err
 	}
-	_, err = writer.Write(readDockerFile)
+	err = buildContextTar.AddAll(constants.InstallationScriptsDir, true)
```

Comment:
It will create the specified directory in the container. In our case, since I am passing the `scripts` directory, it will create a scripts directory and copy all its files in it. If `includeCurrentFolder` is given false, then it will directly copy the files from the scripts directory to container